core/base/inc/TSigHandling.h

```diff
@@ -0,0 +1,94 @@
+// @(#)root/base:$Id$
+// Author: Fons Rademakers   15/09/95
+
+/*************************************************************************
+ * Copyright (C) 1995-2000, Rene Brun and Fons Rademakers.               *
+ * All rights reserved.                                                  *
+ *                                                                       *
+ * For the licensing terms see $ROOTSYS/LICENSE.                         *
+ * For the list of contributors see $ROOTSYS/README/CREDITS.             *
+ *************************************************************************/
+
+#ifndef ROOT_TSigHandling
+#define ROOT_TSigHandling
+
+
+//////////////////////////////////////////////////////////////////////////
+//                                                                      //
+// TSigHandling                                                         //
+//                                                                      //
+// Abstract base class defining a generic interface to the underlying   //
+// Operating System.                                                    //
+//                                                                      //
+//////////////////////////////////////////////////////////////////////////
+
+#ifndef __CINT__
+#include <stdio.h>
+#include <ctype.h>
+#include <fcntl.h>
+#ifndef WIN32
+#include <unistd.h>
+#endif
+#endif
+
+#ifndef ROOT_TNamed
+#include "TNamed.h"
+#endif
+#ifndef ROOT_TString
+#include "TString.h"
+#endif
+#ifndef ROOT_TInetAddress
+#include "TInetAddress.h"
+#endif
+#ifndef ROOT_TTimer
+#include "TTimer.h"
+#endif
+#ifndef ROOT_ThreadLocalStorage
+#include "ThreadLocalStorage.h"
+#endif
+
+class TSeqCollection;
+class TFdSet;
+class TVirtualMutex;
+
+#ifdef __CINT__
+typedef void *Func_t;
+#else
+typedef void ((*Func_t)());
+#endif
+
+R__EXTERN const char  *gRootDir;
```

Comment:
Rather than global variable, it would be better to have them defined a static variable inside a getter function.

For example a function like;

const char *GetRootDir() {
#ifndef ROOTPREFIX
    static const char *gRootDir =  Getenv("ROOTSYS")? Getenv("ROOTSYS") : "/usr/local/root";
#else 
    static const char *gRootDir = ROOTPREFIX;
#endif
    return gRootDir;
}